src/attributes.jl

```diff
@@ -1148,6 +1160,15 @@ struct NumberOfConstraints{F,S} <: AbstractModelAttribute end
 
 attribute_value_type(::NumberOfConstraints) = Int64
 
+function get_fallback(model::ModelLike, ::NumberOfConstraints{F,S}) where {F,S}
+    if supports_constraint(model, F, S) ||
```

Comment:
```suggestion
    # Throw error here so that we don't return an incorrect value if solvers
    # don't implement this method for constraints they support.
    if supports_constraint(model, F, S) ||
```